doc/man3/SSL_read_early.pod

```diff
@@ -0,0 +1,209 @@
+=pod
+
+=head1 NAME
+
+SSL_set_max_early_data,
+SSL_CTX_set_max_early_data,
+SSL_get_max_early_data,
+SSL_CTX_get_max_early_data,
+SSL_SESSION_get_max_early_data,
+SSL_write_early,
+SSL_read_early,
+SSL_get_early_data_status
+- functions for sending and receiving early data
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ int SSL_CTX_set_max_early_data(SSL_CTX *ctx, uint32_t max_early_data);
+ uint32_t SSL_CTX_get_max_early_data(const SSL_CTX *ctx);
+ int SSL_set_max_early_data(SSL *s, uint32_t max_early_data);
+ uint32_t SSL_get_max_early_data(const SSL_CTX *s);
+ uint32_t SSL_SESSION_get_max_early_data(const SSL_SESSION *s);
+
+ int SSL_write_early(SSL *s, const void *buf, size_t num, size_t *written);
+
+ int SSL_read_early(SSL *s, void *buf, size_t num, size_t *readbytes);
+
+ int SSL_get_early_data_status(const SSL *s);
+
+=head1 DESCRIPTION
+
+These functions are used to send and recieve early data. Early data can be sent
+by the client immediately after its initial ClientHello without having to wait
+for the server to complete the handshake. Early data can only be sent if a
+session has previously been established with the server, and the server is known
+to support it.
+
+Early data has weaker security properties than other data sent over an SSL/TLS
+connection. In particular the data is not forward secret and the server has no
+guarantees that the same early data was not replayed across multiple
+connections. For this reason extreme care should be exercised when using early
+data.
+
+When a server receives early data it may opt to immediately respond by sending
+application data back to the client. Data sent by the server at this stage is
+done before the full handshake has been completed. Specifically the client's
+authentication messages have not yet been received, i.e. the client is
+unauthenticated at this point.
+
+A server or client can determine whether the full handshake has been completed
+or not by calling L<SSL_is_init_finished(3)>.
+
+[[TODO(TLS1.3): The server uses SSL_write_ex()/SSL_write() to send data to an
+unauthenticated client. Should we create a separate function for this to avoid
+accidents??]]
```

Comment:
If we think of the concept as "data before the handshake is complete" then ```SSL_write_early_data``` is good.  And unauth is bad for the reasons Todd said.